server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -270,35 +293,38 @@ public void run() {
 
     startDeadCompactionDetector();
 
-    // ELASTICITY_TODO the main function of the following loop was getting group summaries from
-    // tservers. Its no longer doing that. May be best to remove the loop and make the remaining
-    // task a scheduled one.
-
-    LOG.info("Starting loop to check for compactors not checking in");
-    while (!shutdown) {
-      long start = System.currentTimeMillis();
-
-      long now = System.currentTimeMillis();
-      TIME_COMPACTOR_LAST_CHECKED.forEach((k, v) -> {
-        if ((now - v) > getMissingCompactorWarningTime()) {
-          // ELASTICITY_TODO may want to consider of the group has any jobs queued OR if the group
-          // still exist in configuration
-          LOG.warn("No compactors have checked in with coordinator for group {} in {}ms", k,
-              getMissingCompactorWarningTime());
-        }
-      });
+    startIdleCompactionWatcher();
 
-      long checkInterval = getTServerCheckInterval();
-      long duration = (System.currentTimeMillis() - start);
-      if (checkInterval - duration > 0) {
-        LOG.debug("Waiting {}ms for next group check", (checkInterval - duration));
-        UtilWaitThread.sleep(checkInterval - duration);
-      }
+    try {
+      shutdown.await();
+    } catch (InterruptedException e) {
+      LOG.warn("Interrupted waiting for shutdown latch.", e);
     }
 
     LOG.info("Shutting down");
   }
 
+  private Map<String,List<HostAndPort>> getIdleCompactors() {
+
+    Map<String,List<HostAndPort>> allCompactors = ExternalCompactionUtil.getCompactorAddrs(ctx);
```

Comment:
Changing the list to a set will be much more efficient later on the code when it keep removing things from the set.


```suggestion
    Map<String,List<HostAndPort>> allCompactors = ExternalCompactionUtil.getCompactorAddrs(ctx);
    
     Map<String,Set<HostAndPort>> allCompactors  = new HashMap<>();
    allCompactorsLists.forEach((k,v)->allCompactors.put(k, new HashSet<>(v)));
```

Reply:
Implemented in 4fe0683

Reply:
I also created an issue for 2.1 to look at changing the return type of ExternalCompactionUtil.getCompactorAddrs to benefit all uses.